app/routes/questionnaire.py

```diff
@@ -363,6 +367,7 @@ def relationships(
 @with_session_store
 @with_schema
 def get_thank_you(schema, session_store, questionnaire_store):
+    contextvars.clear_contextvars()
```

Comment:
How come this is now needed here?.

Reply:
But is this not needed, in the before request of each endpoint?. As per the doc, Yuyu linked in https://github.com/ONSdigital/eq-questionnaire-runner/pull/1024#discussion_r1104147878?

Reply:
From looking at the logs and comparing them to what we have on master it seems like it is not needed. structlog's `BoundLogger` has [unbind()](https://www.structlog.org/en/stable/api.html#structlog.BoundLogger.unbind) that we never use, so I didn't add [contextvars. unbind_contextvars()](https://www.structlog.org/en/stable/api.html#structlog.contextvars.unbind_contextvars). Seems like [clear_contextvars()](https://www.structlog.org/en/stable/api.html#structlog.contextvars.clear_contextvars) is only optional, we can add it in the before request handler if we want to "reset the context-local context" [docs](https://www.structlog.org/en/stable/contextvars.html) but do we want to?

Reply:
Need to read up on the usage of `clear_contextvar`, it could be new and might be required. On the surface it may see to work but may have mixed context if we don't implement it. So have a look through the docs/online resource on the usage of it and when it is required.

Reply:
Ok, so I've checked the docs, looked at our implementation and it turns out assertions in our tests were badly written and `clear_contextvars()` was indeed needed at the very initial before_request. Now it does behave as the main branch logs and clears keys/values like `tx_d` before the new request. `clear_contextvars()` is added as described in the docs `The typical use-case for this function is to invoke it early in request- handling code` so now no previous request context is used. When compared to `main` it produces the same outputs.

Reply:
Nice one. If not already, can you ensure this is added to every before request hander. We have multiple blueprints handlers.

Reply:
I am not sure if we can add this in other before request handlers as it clears  context vars,  like `request_id` for `questionnaire request` logger.info for example.

Reply:
Okay ya, that's good, that before request is app level which is call for every endpoint. 👍